HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiPartition.java

```diff
@@ -0,0 +1,110 @@
+ *   LUKS partition                           CA7D7CCB-63ED-4C53-861C-1742536059CC
+ *
+ *   linux commands to check uuids:
+ *       blkid list //unique parition guids
+ *       ls /dev/disk/by-partuuid
+ */
+public class UefiPartition {
+  private UefiGuid partitionTypeGUID = null;
+  private UefiGuid uniquePartitionGUID = null;
+  private String partitionName = "";
+  private String attributes = "";
+
+/**
+ * Processes a UEFI defined partition entry.
+ * @param table byte array holding the partition table.
+ * @throws UnsupportedEncodingException if parsing of the data fails.
+ */
+  public UefiPartition(final byte[] table) throws UnsupportedEncodingException {
+    byte[] partitionGUID = new byte[UefiGuid.getGuidLength()];
+    System.arraycopy(table, 0, partitionGUID, 0, UefiGuid.getGuidLength());
+    partitionTypeGUID = new UefiGuid(partitionGUID);
+    byte[] uniquePartGUID = new byte[UefiGuid.getGuidLength()];
+    System.arraycopy(table, UefiGuid.getGuidLength(), uniquePartGUID, 0, UefiGuid.getGuidLength());
+    uniquePartitionGUID = new UefiGuid(uniquePartGUID);
+    byte[] attribute = new byte[UefiConstants.SIZE_8];
+    System.arraycopy(table, UefiConstants.ATTRIBUTE_LENGTH, attribute, 0, UefiConstants.SIZE_8);
+    attributes = HexUtils.byteArrayToHexString(attribute);
+    byte[] partitionname = new byte[UefiConstants.UEFI_PT_LENGTH];
+    System.arraycopy(table, UefiConstants.PART_NAME_LENGTH, partitionname,
+                                                      0, UefiConstants.UEFI_PT_LENGTH);
+    byte[] pName = convertChar16tobyteArray(partitionname);
+    partitionName = new String(pName, "UTF-8").trim();
+   }
+
+/**
+ * Returns the partition Type GIUD.
+ * @return the partition type GUID.
+ */
+public UefiGuid getPartitionTypeGUID() {
+  return partitionTypeGUID;
+}
+/**
+ * Returns the unique partition GUID.
+ * @return the unique partition GUID.
+ */
+public UefiGuid getUniquePartitionGUID() {
+  return uniquePartitionGUID;
+}
+
+/**
+ * Returns the partition name.
+ * @return the partition name.
+ */
+public String getName() {
+  return partitionName;
+  }
+
+/**
+ * Returns a description of the partition.
+ * @return partition description.
+ */
+public String toString() {
+  String partitionInfo = "";
+  partitionInfo += "     Partition Name        : " + partitionName + "\n";
+  partitionInfo += "     Partition Type GUID   : " + partitionTypeGUID.toString() + "\n";
+  partitionInfo += "     Unique Partition GUID : " + uniquePartitionGUID.toStringNoLookup() + "\n";
+  partitionInfo += "     Attributes            : " + attributes + "\n";
+  return partitionInfo;
+ }
+
+/**
+ * Copies character array to a byte by removing upper byte of character array.
+ * @param data input char array
+ * @return byte array
+ */
+private  byte[] convertChar16tobyteArray(final byte[] data) {
+  byte[] hexdata = new byte[data.length];
+  int j = 0;
+  for (int i = 0; i < data.length; i = i + 2) {
```

Comment:
If you are incrementing by 2 just use +=

Reply:
OK